server.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+
+require('dotenv').load();
+
+const cors = require('cors');
+const morgan = require('morgan');
+const express = require('express');
+const debug = require('debug')('thundercatapp:server');
+
+const thundercatRouter = require('./router/thundercat-router.js');
+
+const app = express();
+
+app.use(cors());
+app.use(morgan('dev'));
+
+app.use(thundercatRouter);
+
+app.use(function(err, req, res, next){
```

Comment:
The error middleware was a little buggy from Duncan's lecture code. Here's the more robust middleware I shared in Slack earlier today.

```
  if (err.name === 'MongoError' && err.message.startsWith('E11000 duplicate')){
    // 409 means conflict im using it for duplicate errors 
    err = createError(409, err.message)
    res.status(err.status).send(err.name)
    next()
    return
  }
```